packages/google_maps_flutter/google_maps_flutter/example/integration_test/google_map_inspector.dart

```diff
@@ -2,87 +2,125 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// TODO(a14n): remove this import once Flutter 3.1 or later reaches stable (including flutter/flutter#106316)
-// ignore: unnecessary_import
-import 'dart:typed_data';
-import 'package:flutter/services.dart';
-import 'package:google_maps_flutter/google_maps_flutter.dart';
-
-/// Inspect Google Maps state using the platform SDK.
-///
-/// This class is primarily used for testing. The methods on this
-/// class should call "getters" on the GoogleMap object or equivalent
-/// on the platform side.
-class GoogleMapInspector {
-  GoogleMapInspector(this._channel);
-
-  final MethodChannel _channel;
-
-  Future<bool?> isCompassEnabled() async {
-    return await _channel.invokeMethod<bool>('map#isCompassEnabled');
+import 'package:google_maps_flutter_platform_interface/google_maps_flutter_platform_interface.dart';
+
+/// A method-channel-based implementation of [GoogleMapsInspectorPlatform], for
+/// use in tests in conjunction with [MethodChannelGoogleMapsFlutter].
+// TODO(stuartmorgan): Move this into the platform implementations when
+// federating the mobile implementations.
```

Comment:
This is here with a TODO instead of being moved to live with the default method channel implementation in the platform interface package because I plan to federate the mobile implementations shortly, and move them to in-package method channel implementations, at which point the default method channel inspector would be unused, but we'd be stuck with the extra copy indefinitely since removing it again would be a breaking change. Leaving it local for now lets us avoid that extra copy.